README.md

```diff
@@ -8,22 +8,845 @@
+    revert InvalidSalt({ emitter: _SELF });
+  } else {
+    // In all other cases, the salt value `salt` is not modified.
+    guardedSalt = salt;
+  }
+}
+```
+
+Please note that when you configure a permissioned deploy protection, you **must** specify whether you want a cross-chain redeploy protection or not (i.e. the 21st byte equals `0x00`). The underlying reason for this logic is to enforce developer explicitness. If you don't configure a cross-chain redeploy protection, i.e. the 21st byte is greater than `0x01`, the function reverts. Furthermore, you can configure solely a cross-chain redeploy protection by setting the first 20 bytes equal to the zero address `0x0000000000000000000000000000000000000000`. The rationale behind this logic is to prevent a pseudo-randomly generated cross-chain redeploy protection. Also in this case, if you don't specify a cross-chain redeploy protection, i.e. the 21st byte is greater than `0x01`, the function reverts. The underlying reason for this logic is as well to enforce developer explicitness.
+
+### Pseudo-Random Salt Value
+
+For developer convenience, the [`CreateX`](./src/CreateX.sol) contract offers several overloaded functions that generate the salt value pseudo-randomly using a diverse selection of block and transaction properties. Please note that this approach does not guarantee true randomness!
+
+The full logic is implemented in the `internal` [`_generateSalt`](./src/CreateX.sol#L965-L991) function:
+
+```solidity
+function _generateSalt() internal view returns (bytes32 salt) {
+  // If you use this function between the genesis block and block number 31, it will return zero.
+  unchecked {
+    salt = keccak256(
+      abi.encode(
+        // We don't use `block.number - 256` (the maximum value on the EVM) to accommodate
+        // any chains that may try to reduce the amount of available historical block hashes.
+        // We also don't subtract 1 to mitigate any risks arising from consecutive block
+        // producers on a PoS chain. Therefore, we use `block.number - 32` as a reasonable
+        // compromise, one we expect should work on most chains, which is 1 epoch on Ethereum
+        // mainnet.
+        blockhash(block.number - 32),
+        block.coinbase,
+        block.number,
+        block.timestamp,
+        block.prevrandao,
+        block.chainid,
+        msg.sender
+      )
+    );
+  }
+}
+```
+
+## Design Principles
+
+- [`CreateX`](./src/CreateX.sol) should cover _most_ but not all contract creation use cases.
+- [`CreateX`](./src/CreateX.sol) should be human-readable and should be simple to understand for readers with low prior experience.
+- [`CreateX`](./src/CreateX.sol) should be maximally secure, i.e. no hidden footguns.
+- [`CreateX`](./src/CreateX.sol) should be trustless.
+- [`CreateX`](./src/CreateX.sol) should be stateless.
+- [`CreateX`](./src/CreateX.sol) should be extensible.
```

Comment:
We say "extensible" a few times but I think it might not be clear what we have in mind by that
```suggestion
- [`CreateX`](./src/CreateX.sol) should be extensible (i.e. it can be used to deploy protocols, within protocols, or to deploy other types of deterministic deployer factories).
```